packages/core/lib/process-request.ts

```diff
@@ -86,18 +148,29 @@ export const processRequest = async <TContext = {}, TRootValue = {}>(
   let operation: OperationDefinitionNode | undefined;
 
   const result = await (async (): Promise<ProcessRequestResult<TContext, TRootValue>> => {
-    const accept = typeof request.headers.get === "function" ? request.headers.get("accept") : (request.headers as any).accept;
-    const isEventStream = accept === "text/event-stream";
+    const accept: unknown = getHeader(request, "accept");
+    const contentType: unknown = getHeader(request, "contentType");
+
+    const rankedProtocols = getRankedProtocols(accept, contentType);
```

Comment:
I wonder if this needs some caching? 
Given a pair of `(accept, contentType)` we can always get the same result from `getRankedProtocols`, right? so maybe this could be cached in some way to avoid running the ranking flow?
(feel free to ignore it it's an overkill :D )

Reply:
Caching would require adding some kind of dependency, which we should avoid. The ranking isnt that expensive.